flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -208,7 +210,8 @@ public void onStart() throws Exception {
 			throw exception;
 		}
 
-		dispatcherBootstrap.initialize(this, this.getRpcService().getScheduledExecutor());
+		this.dispatcherBootstrap = this.dispatcherBootstrapFactory.apply(shutDownFuture::completeExceptionally);
```

Comment:
The reason for this is that this would bubble up to `ClusterEntrypoint` calling `System.exit(RUNTIME_FAILURE_RETURN_CODE);` and I do not know if this will kill that cluster without deleting the HA data. 

If this is this the contract, then I can do it.